salt/orch/update.sls

```diff
@@ -287,6 +287,7 @@ kubelet-setup:
     - sls:
       - kubelet.configure-taints
       - kubelet.configure-labels
+{% if masters|length > 0 or workers|length > 0 %}
```

Comment:
That hides the real intent behind the require. Someone at some point might think, why does this require `update-modules` if it's already requiring what it needs, remove that line and again we have the same problem. I think it's safer to keep the explicit guard.

Reply:
The _remove that line and something bad will happen_ argument can be used for anything :-). No lines in this file can be removed without carefully thinking about the consequences.

Besides that, when no masters or minions are available, `update-modules` is the real previous stage, so the only thing you are doing by requiring it is setting the right order. You would need to add it in your code anyway or the order would not be fully specified.

And, when there are masters and/or minions, a `require` for `update-modules` would specify a requirement for a stage that has already been done, so it would have no effect.

Reply:
Done.